omnigibson/object_states/slicer_active.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+from omnigibson.macros import create_module_macros
+from omnigibson.object_states.object_state_base import AbsoluteObjectState
+from omnigibson.object_states.aabb import AABB
+from omnigibson.object_states.contact_bodies import ContactBodies
+from omnigibson.object_states.update_state_mixin import UpdateStateMixin
+import omnigibson as og
+
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+class SlicerActive(AbsoluteObjectState, UpdateStateMixin):
+    @classmethod
+    def get_dependencies(cls):
+        deps = super().get_dependencies()
+        deps.add(AABB)
+        deps.add(ContactBodies)
+        return deps
+
+    def __init__(self, obj):
+        super(SlicerActive, self).__init__(obj)
+
+        self._previously_touching_sliceable = False
+        self.value = True
+        self.delay_counter = 0
+        self.delay_steps = 7
+
+    def _get_value(self):
+        return self.value
+
+    def _set_value(self, new_value):
+        self.value = new_value
+        return True
+    
+    def _currently_touching_sliceable(self):
+        contact_links = self.obj.states[ContactBodies].get_value()
+        contact_link_prim_paths = {contact_link.prim_path for contact_link in contact_links}
+        for prim_path in contact_link_prim_paths:
+            obj_prim_path, _ = prim_path.rsplit("/", 1)
+            candidate_obj = og.sim.scene.object_registry("prim_path", obj_prim_path, None)
+            if candidate_obj is None:
+                continue
+            if hasattr(candidate_obj, '_abilities') and "sliceable" in candidate_obj._abilities:
+                return True
+        return False
+
+    def _update(self):
+        # If we were slicing in the past step, deactivate now.
+        if self._previously_touching_sliceable:
+            self.set_value(False)
+            self.delay_counter = 0  # Reset the counter when we stop touching a sliceable object
+
+        # Are we currently touching any sliceables?
+        currently_touching_sliceable = self._currently_touching_sliceable()
+
+        # If our value is False, we need to consider reverting back.
+        if not self.value:
+            # If we are not touching any sliceable objects, we can revert to True.
+            if not currently_touching_sliceable:
+                self.delay_counter += 1  # Increment the counter for each step we're not touching a sliceable object
+                if self.delay_counter >= self.delay_steps:  # Check if the counter has reached the delay
```

Comment:
Note that delay_counter gets incremented once every rendering timestep which is a parameter. By default (at 1/60sec) it will happen at 0.1s but this parameter can be much higher. Perhaps it's best to specify the macro as a number of seconds (e.g. 0.1), then compute it by `min(1, m.REACTIVATION_DELAY / og.sim.rendering_dt)` to decide how many steps to wait.

Reply:
Also let's just reintroduce that cooldown to be like 0.5 sec for now imo. @ChengshuLi FYI

Reply:
Hi @cgokmen, just making sure that I understand this correctly: 
For example, if we set `REACTIVATION_DELAY` to be 0.1 and `og.sim.get_rendering_dt()` is now 0.016, that would give us about 6 steps/frames to wait. If we then take `min(1, m.REACTIVATION_DELAY / og.sim.get_rendering_dt())`, wouldn't that always give us 1? I guess for the min cap, did you mean something like 10 instead?